cg/services/order_validation_service/validators/data_validators.py

```diff
@@ -0,0 +1,45 @@
+from cg.services.order_validation_service.models.errors import (
+    CustomerCannotSkipReceptionControlError,
+    CustomerDoesNotExistError,
+    UserNotAssociatedWithCustomerError,
+    ValidationError,
+)
+from cg.services.order_validation_service.models.order import Order
+from cg.store.store import Store
+
+
+def validate_user_belongs_to_customer(
+    order: Order, store: Store, **kwargs
+) -> list[ValidationError]:
+    has_access: bool = store.is_user_associated_with_customer(
+        user_id=order.user_id,
+        customer_internal_id=order.customer_internal_id,
+    )
+
+    errors: list[ValidationError] = []
+    if not has_access:
+        error = UserNotAssociatedWithCustomerError()
+        errors.append(error)
+    return errors
+
+
+def validate_customer_can_skip_reception_control(
+    order: Order, store: Store, **kwargs
+) -> list[ValidationError]:
+    errors: list[ValidationError] = []
+
+    if not order.skip_reception_control:
+        return errors
+
+    if not store.is_customer_trusted(order.customer_internal_id):
+        error = CustomerCannotSkipReceptionControlError()
+        errors.append(error)
+    return errors
+
+
+def validate_customer_exists(order: Order, store: Store, **kwargs) -> list[ValidationError]:
+    errors: list[ValidationError] = []
+    if not store.customer_exists(order.customer_internal_id):
```

Comment:
The error should be raised from the CRUD and caught here. It is the database that tells you the lookup was wrong.

Reply:
For this function, I prefer the return to be a bool instead of an exception

Reply:
Yes, I was unclear. I think it might be generally better to raise an error in the "get_x_by_unique_identifier" to raise an error when it returns nothing. Meaning we shouldn't need an "customer_exists" function in the crud. You either get the entry or an error. What do you think?

Reply:
I don't  think there is anything wrong with having an explicit method for an existence check. I do agree that we should consider a uniform design for our data access layer when you attempt to retrieve something and it does not exist. 

I do think I prefer having the optional return type for our complex queries over raising an exception, but I'm not sure. Maybe this is a discussion that is better suited elsewhere? I feel like it is out of scope for this PR at least.